app/main/views/service_updates.py

```diff
@@ -54,33 +56,24 @@ def service_status_update_audits(day=None, page=1):
 @login_required
 @role_required('admin', 'admin-ccs-category')
 def service_update_audits():
-    form = ServiceUpdateAuditEventsForm(request.args, csrf_enabled=False)
-
-    if not form.validate():
-        return render_template(
-            "service_update_audits.html",
-            today=datetime.utcnow().strftime(DATETIME_FORMAT),
-            acknowledged=form.default_acknowledged(),
-            audit_events=[],
-            form=form
-        ), 400
-
-    audit_events = data_api_client.find_audit_events(
+    audit_events = list(data_api_client.find_audit_events_iter(
```

Comment:
I think we're largely on the same page on this. The API endpoint was never meant to support this use case, so building this directly on top of it doesn't feel like a good approach (for all the reasons you've mentioned). Same goes for https://github.com/alphagov/digitalmarketplace-admin-frontend/pull/281/files, which is trying to collect events for an individual service.

I think as an alternative, having a separate collection of API endpoints (eg `/service-updates` would be a simple option, but we can think of what it would take to build a more general one to cover other types of updates as well) with a clear interface would go a long way towards addressing this. Or we can look into extending the interface of the existing endpoint if it fits better.

I'm not particularly committed to the idea of using `ArchivedServices` or denormalizing this data either. I'd be more than happy if we can get it to work with the existing data model. But as you know we don't have a lot of examples of similar queries against large tables or a robust set of performance metrics we can compare against, so there are a lot of assumptions going into this both in terms of performance of the admin page / API endpoint / DB queries themselves and of their effect on the rest of the apps. This doesn't mean that we can't do this, but if we're going to then we do need to explicitly highlight the risks and our expectations, make a plan to test, measure and validate them before we release and ensure that we'll be able (including having time in the future sprints) to fix any issues that arise if our assumptions don't hold.

I'm going to schedule a chat on Monday so we can all talk about this.